lightning/src/ln/msgs.rs

```diff
@@ -570,6 +570,77 @@ pub struct ChannelUpdate {
 	pub contents: UnsignedChannelUpdate,
 }
 
+/// A query_channel_range message that can be sent or received from a peer
+#[derive(Clone, Debug)]
+pub struct QueryChannelRange {
+	/// The genesis hash of the blockchain being queried
+	pub chain_hash: BlockHash,
+	/// The height of the first block for the channel UTXOs being queried
+	pub first_blocknum: u32,
+	/// The number of blocks to include in the query results
+	pub number_of_blocks: u32,
+}
+
+/// A reply_channel_range message that can be sent or received from a peer.
+/// Multiple reply_channel_range messages can be sent in reply to a single
+/// query_channel_range message. The query recipient makes a best effort
+/// to respond based on their local network view which may not be
+/// a perfect view of the network.
+#[derive(Clone, Debug)]
+pub struct ReplyChannelRange {
+	/// The genesis hash of the blockchain being queried
+	pub chain_hash: BlockHash,
+	/// The height of the first block in the range of the reply
+	pub first_blocknum: u32,
+	/// The number of blocks included in the range of the reply
+	pub number_of_blocks: u32,
+	/// Indicates if the query recipient maintains up-to-date channel
+	/// information for the chain_hash
+	pub full_information: bool,
+	/// The short_channel_ids in the channel range
+	pub short_channel_ids: Vec<u64>,
+}
+
+/// A query_short_channel_ids that can be seent or received from a peer
+#[derive(Clone, Debug)]
+pub struct QueryShortChannelIds {
+	/// The genesis hash of the blockchain being queried
+	pub chain_hash: BlockHash,
+	/// The short_channel_ids that are being queried
+	pub short_channel_ids: Vec<u64>,
+}
+
+/// A reply_short_channel_ids_end message that can be sent or received
+/// from a peer. This message is sent as a reply to a query_short_channel_ids
+/// message. The query recipient makes a best effort to respond based on
+/// their local network view which may not be a perfect view of the network.
+#[derive(Clone, Debug)]
+pub struct ReplyShortChannelIdsEnd {
+	/// The genesis hash of the blockchain that was queried
+	pub chain_hash: BlockHash,
+	/// Indicates if the query recipient maintains up-to-date channel
+	/// information for the chain_hash
+	pub full_information: bool,
+}
+
+/// A gossip_timestamp_filter message is used by a node to request
+/// gossip relay for messages in the requested time range when the
+/// gossip_queries feature has been negotiated.
+#[derive(Clone, Debug)]
+pub struct GossipTimestampFilter {
+	/// The genesis hash of the blockchain for channel and node information
+	pub chain_hash: BlockHash,
+	/// The starting unix timestamp
+	pub first_timestamp: u32,
+	/// The range of information in seconds
+	pub timestamp_range: u32,
+}
+
+/// Encoding type for data compression of collections in gossip queries
+enum EncodingType {
```

Comment:
Maybe mark explicitly somewhere we don't support spec encoding-type of 1 for zlib serialization.

Reply:
Added comment about not supporting zlib